### ISSUE TITLE
[Enhancement]: Make Standalone.IL2CPP target Il2CppInterop, not Unhollower

### ISSUE THREAD

**Issue by slxdy** [yukieiji/UnityExplorer#12]
### Describe the new feature or enhancement

Pretty much what the title says. Reason is self-explanatory

**Comment by yukieiji** [comment]
Will this work?
[UnityExplorer.Standalone.IL2CPP.CoreCLR.zip](https://github.com/yukieiji/UnityExplorer/files/14729063/UnityExplorer.Standalone.IL2CPP.CoreCLR.zip)

**Comment by slxdy** [comment]
I will check it out. I'm working on my own mod loader, so imma let you know if it works

**Comment by slxdy** [comment]
@yukieiji Can you also make the `OnLog` event use its own LogType enum instead of using UnityEngine's one? Saves me an assembly reference 😅

**Comment by slxdy** [comment]
Anyway, I managed to load UnityExplorer, except it doesn't generate it's UI for whatever reason? There is one type cache error tho, but not sure if it's related.

Log:
```
[18:23:33.898][Mod Loader] Mod loaded: [ UnityExplorer, ..\..\GlobalMods\UnityExplorer\UnityExplorer.dll ]
[18:23:35.308][UnityExplorer] [UniverseLib] Can't cache type named ParticlesQualityConfig Error: System.TypeLoadException: GenericArguments[0], 'EnumPublicSealedvaHIMELOVE5vUnique', on 'Object2PublicAbstractLi1ObDi2AcObLi1BoUnique`1[T]' violates the constraint of type parameter 'T'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[18:23:41.493][UnityExplorer] [UniverseLib] Initialized Legacy Input support
```

Mod implementation:
```cs
namespace UnityExplorer;

public class Main : IMod
{
    // Register hooks and events in this function
    public static void Initialize()
    {
        ExplorerStandalone.CreateInstance(HandleLog);
    }

    private static void HandleLog(string arg1, LogType type)
    {
        ModLogger.Log(arg1, type switch
        {
            LogType.Warning => "yellow",
            LogType.Error or LogType.Exception or LogType.Assert => "red",
            _ => null
        });
    }
}
```

**Comment by slxdy** [comment]
Ok, so I fixed the cache error, that was on my side. I was initializing UE too early, so I decided to initialize it after the first "Start" function call in a MonoBehaviour. But the UI problem still persists

New log:
```
[18:33:33.177][Mod Loader] Mod loaded: [ UnityExplorer, ..\..\GlobalMods\UnityExplorer\UnityExplorer.dll ]
[18:33:39.984][UnityExplorer] [UniverseLib] Initialized Legacy Input support
```

**Comment by slxdy** [comment]
I also manually specified the path to the proxy modules, but that didn't seem to do much

**Comment by slxdy** [comment]
After using a debugger, this one might actually be on my part as well. Imma dig into it
![image](https://github.com/yukieiji/UnityExplorer/assets/61495410/454f6147-e910-46f6-b99e-08523fd4da04)

**Comment by yukieiji** [comment]
> Can you also make the OnLog event use its own LogType enum instead of using UnityEngine's one? Saves me an assembly reference

This's outside the scope of this ticket, as it will need to be discussed and tested separately(I will correspond separately.)

> Anyway, I managed to load UnityExplorer, except it doesn't generate it's UI for whatever reason? There is one type cache error tho, but not sure if it's related.
Ok, so I fixed the cache error, that was on my side. I was initializing UE too early, so I decided to initialize it after the first "Start" function call in a MonoBehaviour. But the UI problem still persists
I also manually specified the path to the proxy modules, but that didn't seem to do much

The Il2CppInterop-supported version I gave you was created by changing reference without changing any UE code.
Therefore, it seems unlikely that it is a bug on the UE side.

I will fine-tune and officially release the product as it seems to work fine.
